neuron/neuron/checkpoint.py

```diff
@@ -165,71 +166,82 @@ def compare_checkpoints(contest: Contest, repository: str) -> CheckpointBenchmar
+                f"and similarity {similarity}"
+            )
 
-        if generated:
-            average_time = (average_time * generated + generation.generation_time) / (generated + 1)
-        else:
-            average_time = generation.generation_time
+            if generated:
+                average_time = (average_time * generated + generation.generation_time) / (generated + 1)
+            else:
+                average_time = generation.generation_time
 
-        average_similarity = (average_similarity * generated + similarity) / (generated + 1)
+            average_similarity = (average_similarity * generated + similarity) / (generated + 1)
 
-        if average_time < baseline_average * 1.0625:
-            # So far, the average time is better than the baseline, so we can continue
-            continue
+            if average_time < baseline_average * 1.0625:
+                # So far, the average time is better than the baseline, so we can continue
+                continue
 
-        needed_time = (baseline_average * SAMPLE_COUNT - generated * average_time) / remaining
+            needed_time = (baseline_average * SAMPLE_COUNT - generated * average_time) / remaining
 
-        if needed_time < average_time * 0.75:
-            # Needs %33 faster than current performance to beat the baseline,
-            # thus we shouldn't waste compute testing farther
-            failed = True
-            bt.logging.info("Current average is 75% of the baseline average")
-            break
+            if needed_time < average_time * 0.75:
+                # Needs %33 faster than current performance to beat the baseline,
+                # thus we shouldn't waste compute testing farther
+                failed = True
+                bt.logging.info("Current average is 75% of the baseline average")
+                break
 
-        if average_similarity < 0.85:
-            # Deviating too much from original quality
-            bt.logging.info("Too different from baseline, failing")
-            failed = True
-            break
+            if average_similarity < 0.85:
+                # Deviating too much from original quality
+                bt.logging.info("Too different from baseline, failing")
+                failed = True
+                break
 
-    bt.logging.info(
-        f"Tested {i + 1} samples, "
-        f"average similarity of {average_similarity}, "
-        f"and speed of {average_time}"
-    )
+        bt.logging.info(
+            f"Tested {i + 1} samples, "
+            f"average similarity of {average_similarity}, "
+            f"and speed of {average_time}"
+        )
 
     return CheckpointBenchmark(
         baseline_average,
         average_time,
         average_similarity,
         failed,
     )
+
+
+@contextmanager
+def load_pipeline(contest: Contest, repository: str | None = None):
+    try:
+        pipeline = contest.load(repository)
+        yield pipeline
+    finally:
+        del pipeline
```

Comment:
No need for explicitly deleting this variable, which is gone when this function exits anyways.